content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md

```diff
@@ -24,26 +26,33 @@ CNI specification (plugins can be compatible with multiple spec versions).
+For specific information about how a Container Runtime manages the CNI plugins, see the documentation for that Container Runtime, for example:
+- [containerd](https://github.com/containerd/containerd/blob/main/script/setup/install-cni)
+- [CRI-O](https://github.com/cri-o/cri-o/blob/main/contrib/cni/README.md)
 
-By default if no kubelet network plugin is specified, the `noop` plugin is used, which sets `net/bridge/bridge-nf-call-iptables=1` to ensure simple configurations (like Docker with a bridge) work correctly with the iptables proxy.
+For specific information about how to install and manage a CNI plugin, see the documentation for that plugin.
 
-### CNI
+## Network Plugin Requirements
+
+The plugin may also need specific support for kube-proxy.  The iptables proxy obviously depends on iptables, and the plugin may need to ensure that container traffic is made available to iptables.  For example, if the plugin connects containers to a Linux bridge, the plugin must set the `net/bridge/bridge-nf-call-iptables` sysctl to `1` to ensure that the iptables proxy functions correctly.  If the plugin does not use a Linux bridge (but instead something like Open vSwitch or some other mechanism) it should ensure container traffic is appropriately routed for the proxy.
 
-The CNI plugin is selected by passing Kubelet the `--network-plugin=cni` command-line option.  Kubelet reads a file from `--cni-conf-dir` (default `/etc/cni/net.d`) and uses the CNI configuration from that file to set up each pod's network.  The CNI configuration file must match the [CNI specification](https://github.com/containernetworking/cni/blob/master/SPEC.md#network-configuration), and any required CNI plugins referenced by the configuration must be present in `--cni-bin-dir` (default `/opt/cni/bin`).
+By default if no kubelet network plugin is specified, the `noop` plugin is used, which sets `net/bridge/bridge-nf-call-iptables=1` to ensure simple configurations (like Docker with a bridge) work correctly with the iptables proxy.
 
-If there are multiple CNI configuration files in the directory, the kubelet uses the configuration file that comes first by name in lexicographic order.
+### Loopback CNI
 
-In addition to the CNI plugin specified by the configuration file, Kubernetes requires the standard CNI [`lo`](https://github.com/containernetworking/plugins/blob/master/plugins/main/loopback/loopback.go) plugin, at minimum version 0.2.0
+In addition to the CNI plugin installed on the nodes, Kubernetes requires the standard CNI [`lo`](https://github.com/containernetworking/plugins/blob/master/plugins/main/loopback/loopback.go) plugin, at minimum version 0.2.0
```

Comment:
> or by other more direct means

I'm not sure what this part means - do you have an example (or simple explanation) of what this might look like?  Might be good to link to an example or other explanation (or include that here).

Reply:
Instead of letting the plugin do the work.. you do the work of the plugin. Implementing / enabling the loopback interface is a small amount of well known code. CRI-O, for example, does not call the loopback plugin. Instead it follows this path: https://github.com/cri-o/ocicni/blob/5b6be5e9ec4bd52fd078580dbb5a367449a717c6/pkg/ocicni/util_linux.go#L91

Reply:
this may be TMI.. Feel free thin it up.

Reply:
I may have made it longer!  Let me know what you think - modified in https://github.com/kubernetes/website/pull/33870/commits/dc6d1fb667fac4019de02eacc0cc1fa5b9e0647f.